backend/web/workflows/linux_qc_patching_prep.py

```
@@ -0,0 +1,766 @@
+            raise HTTPException(
+                status_code=401,
+                detail="Not authenticated. Please authenticate first via /auth/authenticate",
+            )
+
+    try:
+        return get_provider(provider_name, credentials)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail=str(e),
+        )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)